File: rust/242_valid_anagram.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }

        let mut letter_freq = [0; 26];
        s.bytes()
            .for_each(|c| letter_freq[(c - b'a') as usize] += 1);
        t.bytes()
            .for_each(|c| letter_freq[(c - b'a') as usize] -= 1);

        letter_freq.into_iter().all(|&x| x == 0)
    }
}

File: cpp/242_valid_anagram.cpp
#include <array>
#include <string>

using namespace std;

class Solution {
 public:
  constexpr int toIndex(char c) { return c - 'a'; }
  bool isAnagram(string s, string t) {
    if (s.size() != t.size()) {
      return false;
    }
    array<int, 26> letter_freq;
    letter_freq.fill(0);
    for (const auto &c : s) {
      letter_freq[toIndex(c)]++;
    }

    for (const auto &c : t) {
      if (letter_freq[toIndex(c)] == 0) {
        return false;
      }
      letter_freq[toIndex(c)]--;
    }
    return true;
  }
};
